src/read_pool.rs

```diff
@@ -0,0 +1,158 @@
+use futures::sync::oneshot;
+use futures::{future, Future};
+use futures03::prelude::*;
+use kvproto::kvrpcpb::CommandPri;
+use std::future::Future as StdFuture;
+use tikv_util::future_pool::{self, FuturePool};
+use yatp::pool::{Local, Runner};
+use yatp::queue::Extras;
+use yatp::task::future::{Runner as FutureRunner, TaskCell};
+use yatp::Remote;
+
+use crate::storage::kv::{destroy_tls_engine, set_tls_engine, Engine};
+
+#[derive(Clone)]
+pub enum ReadPool {
+    FuturePools {
+        read_pool_high: FuturePool,
+        read_pool_normal: FuturePool,
+        read_pool_low: FuturePool,
+    },
+    Yatp(Remote<TaskCell>),
+}
```

Comment:
Would `enum` perform better than `trait`  or `dyn impl`?

Reply:
Yes. `enum` performs better than dynamic dispatch. Static dispatch through trait is zero-cost but makes code too complex.